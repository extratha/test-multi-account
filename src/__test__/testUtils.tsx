import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
