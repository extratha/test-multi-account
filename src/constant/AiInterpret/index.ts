export const FIELD_NAME = {
  GENDER: "gender",
  STOOL_COLOR: "stool_color",
  STOOL_WBC: "stool_wbc",
  STOOL_RBC: "stool_rbc",
  STOOL_PARASITE: "stool_parasite",
  STOOL_OCCULT_BLOOD: "stool_occult_blood",
  STOOL_CULTURE: "stool_culture",
  URINE_COLOR: "urine_color_value",
  URINE_SP_GR_VALUE: "urine_sp_gr_value",
  URINE_PH_VALUE: "urine_ph_value",
  URINE_WBC_VALUE: "urine_wbc_value",
  URINE_PROTEIN_VALUE: "urine_protein_value",
  URINE_GLUCOSE_VALUE: "urine_glucose_value",
  URINE_ERYTHROCYTE_VALUE: "urine_erythrocyte_value",
  URINE_RBC_VALUE: "urine_rbc_value",
};
