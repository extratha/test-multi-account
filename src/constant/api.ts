
export const API = {
  PATH: {
    login: 'auth/login',
    setNewPassword: 'auth/reset-password',
    changePassword: 'auth/change-password',
    forgotPassword: 'auth/forgot-password',
    exampleDataList: 'lab/examples',
  },
  HTTP_METHODS: {
    GET: 'GET',
    POST: 'POST',
    PUT: 'PUT',
    PATCH: 'PATCH',
    DELETE: 'DELETE',
  },
  ERROR: {
    UNAUTHENTICATION: {
      MESSAGE: 'unauthenticated',
      CODE: 401,
    },
    login: {
      INVALID_CREDENTIAL: {
        MESSAGE: 'invalid email or password',
        CODE: 400,
      },
    }
  },
};
