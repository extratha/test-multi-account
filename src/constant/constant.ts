export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};
