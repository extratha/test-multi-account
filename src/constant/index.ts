export const ENV = {
  BASE_API_URL: process.env.NEXT_PUBLIC_BASE_API_URL || "MOCK_NEXT_PUBLIC_BASE_API_URL",
  FIREBASE_API_KEY: process.env.NEXT_PUBLIC_FIREBASE_API_KEY || "MOCK_NEXT_PUBLIC_FIREBASE_API_KEY",
  FIREBASE_AUTH_DOMAIN: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN || "MOCK_NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN",
  FIREBASE_PROJECT_ID: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_PROJECT_ID",
  FIREBASE_STORAGE_BUCKET:
    process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET || "MOCK_NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET",
  FIREBASE_MESSAGING_SENDER_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID",
  FIREBASE_APP_ID: process.env.NEXT_PUBLIC_FIREBASE_APP_ID || "MOCK_NEXT_PUBLIC_FIREBASE_APP_ID",
  FIREBASE_MEASUREMENT_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID",
};

export const SESSION = {
  ACCESS_TOKEN: "access-token",
  REFRESH_TOKEN: "refresh-token",
  RESET_PASSWORD_TOKEN: "reset-password-token",
  X_ROBOT_TOKEN: "x-robot-token",
};

export const MENU_CONDITION = {
  LOGOUT: "logout",
};

export const NAVIGATION = {
  NOT_FOUND: "/not-found",
  LOGIN: "/login",
  HOME: "/home",
  FORGET_PASSWORD: "/forget-password",
  SET_NEW_PASSWORD: "/set-new-password",
  CONSENT_TERMS_CONDITIONS: "/consents/terms-conditions",
  CONSENT_PRIVACY_POLICIES: "/consents/privacy-policies",
  AI_INTERPRET_RESULT: "/ai-interpret/interpret",
  AI_INTERPRET_TRY_EXAMPLE_DATA: "/ai-interpret/try-example-data",
  AI_INTERPRET_TRY_INPUT_DATA: "/ai-interpret/try-input-data",
  SYMPTOM_CHECKER: "/symptom-checker",
  SETTINGS_TERMS_AND_CONDITIONS: "/settings/terms-and-conditions",
  SETTINGS_PRIVACY_POLICY: "/settings/privacy-policy",
  PUBLIC_TERMS_AND_CONDITIONS: "/public/consents/terms-conditions",
  PUBLIC_PRIVACY_POLICY: "/public/consents/privacy-policies",
};

export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};

export const GROUP_NAME = {
  GENERAL_CHECK_UP: "generalCheckUp",
  HEMATOLOGY_BG: "hematologyBG",
  HEMATOLOGY_CBC: "hematologyCBC",
  LIPID_PROFILE: "lipidProfile",
  BLOOD_CHEMISTRY: "bloodChemistry",
  RENAL_FUNCTION_TEST: "renalFunctionTest",
  LIVER_FUNCTION_TEST: "liverFunctionTest",
  URINE_ANALYSIS: "urineAnalysis",
  STOOL_EXAMINATION: "stoolExamination",
  PAP_SMEAR: "papSmear",
  TUMOR_MARKER: "tumorMarker",
};

export const GENERAL_CHECK_UP = {
  AGE: "age",
  GENDER: "gender",
  BODY_WEIGHT: "body_weight",
  BODY_HEIGHT: "body_height",
  WAIST_LINE: "waist_line",
  BLOOD_PRESSURE_SYSTOLIC: "blood_pressure_systolic",
  BLOOD_PRESSURE_DIASTOLIC: "blood_pressure_diastolic",
  PULSE: "pulse",
  RESPIRATORY_RATE: "respiratory_rate",
  TEMPERATURE: "temperature",
};

export const HEMATOLOGY_BLOOD = {
  BLOOD_GROUP: "blood_group",
  BLOOD_GROUP_RH: "blood_group_rh",
};

export const HEMATOLOGY_BLOOD_GROUP = {
  HEMATOLOGY_BLOOD: "bloodGroup",
};

export const HEMATOLOGY_CBC = {
  HB: "hb_value",
  HCT: "hct_value",
  RBC_MORPHOLOGY: "rbc_morphology_value",
  MCV: "mcv_value",
  MCH: "mch_value",
  MCHC: "mchc_value",
  RDW: "rdw_value",
  WBC: "wbc_value",
  NEUTROPHIL: "neutrophil_value",
  LYMPHOCYTE: "lymphocyte_value",
  MONOCYTE: "monocyte_value",
  EOSINOPHIL: "eosinophil_value",
  BASOPHIL: "basophil_value",
  PLATELET_COUNT: "platelet_count_value",
};

export const HEMATOLOGY_CBC_GROUP = {
  HEMOGLOBIN: "hemoglobin",
  WHITE_BLOOD: "whiteBlood",
  RED_BLOOD: "redBlood",
};

export const INTERPRET_STATUS = {
  PENDING: "pending",
  SUCCESS: "success",
  FAILED: "failed",
};

export const ASR_SERVICE = {
  ORDER_ASR: "order-asr",
  SOAP_ASR: "soap-asr",
};

export const ASR_SERVICE_KEY: Record<string, string> = {
  [ASR_SERVICE.ORDER_ASR]: "orderAsr",
  [ASR_SERVICE.SOAP_ASR]: "soapAsr",
};

export const CONSENT_TYPE = {
  TERMS_CONDITIONS: "terms-conditions",
  PRIVACY_POLICIES: "privacy-policies",
};

export const SUBMIT_CONSENT_TYPE = {
  [CONSENT_TYPE.TERMS_CONDITIONS]: "terms",
  [CONSENT_TYPE.PRIVACY_POLICIES]: "privacy_policies",
};
