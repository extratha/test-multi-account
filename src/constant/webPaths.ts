import SetNewPasswordPage from "@/app/[locale]/set-new-password/page";

export const webPaths = {
  landing: '/landing',
  home: '/home',
  forgetPassword: '/forget-password',
  setNewPassword: '/set-new-password',
};
