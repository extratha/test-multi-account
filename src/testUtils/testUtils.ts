import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
