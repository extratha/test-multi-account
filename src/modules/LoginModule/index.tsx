import VisibilityIcon from "@mui/icons-material/Visibility";
import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";
import { IconButton, Stack, styled, Typography } from "@mui/material";
import Link from "next/link";
import { useRouter } from "next/navigation";
import { useEffect, useState } from "react";
import { Controller, SubmitHandler, useForm } from "react-hook-form";

import { submitLogin } from "@/api/apiUnauthorize";
import { SubmitButtonStyle } from "@/components/Button/styled";
import FullScreenLoading from "@/components/Loading/FullScreenLoading";
import { NAVIGATION, SESSION } from "@/constant";
import useFieldValidation from "@/hooks/useFieldValidation";
import useTranslation from "@/locales/useLocale";
import { useUserProfileStore } from "@/store";
import { getVersion } from "@/utils/common";
import { validateEmail, validatePassword } from "@/utils/validation";
import { CustomTextField } from "./styled";

export interface LoginFormValues {
  email: string | null;
  password: string | null;
}

const Container = styled(Stack)({
  width: "100%",
  maxWidth: "66%",
  margin: "auto 0px",
  padding: "16px",
});

const ForgetPassword = styled(Typography)({
  paddingTop: "32px",
  textDecoration: "underline",
  textAlign: "center",
  cursor: "pointer",
});

const ConsentFooter = styled(Stack)({
  justifyContent: "center",
  padding: "16px 0px 32px",
  "& .MuiTypography-root": {
    cursor: "pointer",
    minWidth: "160px",
    padding: "8px 16px",
    textAlign: "center",
  },
  "& a": {
    textDecoration: "none",
    color: "inherit",
  },
});

const LoginModule = () => {
  const { translation } = useTranslation();
  const router = useRouter();
  const { resetUserProfile, setUserProfile } = useUserProfileStore();
  const [isDisableSubmit, setIsDisableSubmit] = useState(true);
  const [showPassword, setShowPassword] = useState(false);
  const { control, setError, handleSubmit } = useForm<LoginFormValues>();
  const [errorMessage, setErrorMessage] = useState("");
  const [isSubmitting, setIsSubmitting] = useState(false);

  const EMAIL_FIELD_NAME = "email";
  const PASSWORD_FIELD_NAME = "password";
  const {
    isValid: isEmailValid,
    value: emailValue,
    onChange: onEmailChange,
    onBlur: onEmailBlur,
    isDirty: isEmailDirty,
    ref: emailRef,
  } = useFieldValidation(control, EMAIL_FIELD_NAME, validateEmail, setError);

  const {
    isValid: isPasswordValid,
    value: passwordValue,
    onChange: onPasswordChange,
    onBlur: onPasswordBlur,
    isDirty: isPasswordDirty,
    ref: passwordRef,
  } = useFieldValidation(control, PASSWORD_FIELD_NAME, validatePassword, setError);

  useEffect(() => {
    if (!isEmailDirty || !isPasswordDirty) return;
    if (!(isEmailValid && isPasswordValid)) {
      setIsDisableSubmit(true);
    } else {
      setIsDisableSubmit(false);
    }
  }, [emailValue, passwordValue]);

  const handleToggleShowPassword = () => {
    setShowPassword((prev) => !prev);
  };

  // TODO: unit after login
  const onSubmit: SubmitHandler<LoginFormValues> = async () => {
    try {
      setIsSubmitting(true);
      setErrorMessage("");
      resetUserProfile();

      const response = await submitLogin({ email: emailValue || "", password: passwordValue || "" });
      const { accessToken, refreshToken, user, userProfile } = response.data;

      localStorage.setItem(SESSION.ACCESS_TOKEN, accessToken);
      localStorage.setItem(SESSION.REFRESH_TOKEN, refreshToken);

      setUserProfile(userProfile);

      if (user.passwordChanged) {
        router.replace(NAVIGATION.CONSENT_TERMS_CONDITIONS);
      } else {
        router.replace(NAVIGATION.SET_NEW_PASSWORD);
      }
    } catch (error) {
      setErrorMessage(error?.response?.data?.message || translation("login.text.error.invalidEmailOrPassword"));
      setIsSubmitting(false);
    }
  };

  return (
    <>
      {isSubmitting && <FullScreenLoading />}
      <Stack flex="1">
        <Stack flex="1" alignItems="center">
          <Container>
            <Stack spacing="8px" marginBottom="32px">
              <Typography variant="titleBold" color="text.medium">
                {translation("login.text.login")}
              </Typography>
              <Typography variant="headerBold">{translation("login.text.carivaPlayground")}</Typography>
              <Typography variant="labelExtraSmall" color="text.disabled">
                {translation("login.text.version", { version: getVersion() })}
              </Typography>
            </Stack>
            <form onSubmit={handleSubmit(onSubmit)}>
              <Controller
                name={EMAIL_FIELD_NAME}
                control={control}
                render={({ fieldState: { error } }) => (
                  <Stack direction="column" spacing={2} mt={1}>
                    <CustomTextField
                      fullWidth
                      data-testid="email-input"
                      id={EMAIL_FIELD_NAME}
                      placeholder={!emailValue ? translation("login.text.placeholder.email") : ""}
                      name={EMAIL_FIELD_NAME}
                      value={emailValue ?? ""}
                      onChange={onEmailChange}
                      onBlur={onEmailBlur}
                      inputRef={emailRef}
                      InputLabelProps={{ shrink: true }}
                    />
                    <Typography
                      data-testid="email-error"
                      variant="labelExtraSmallMedium"
                      textAlign="left"
                      color="error"
                    >
                      {error?.message || ""}
                    </Typography>
                  </Stack>
                )}
              />
              <Controller
                name={PASSWORD_FIELD_NAME}
                control={control}
                render={({ fieldState: { error } }) => (
                  <Stack direction="column" spacing={2} mt={1}>
                    <CustomTextField
                      fullWidth
                      name={PASSWORD_FIELD_NAME}
                      type={showPassword ? "text" : "password"}
                      data-testid="password-input"
                      id={PASSWORD_FIELD_NAME}
                      placeholder={!passwordValue ? translation("login.text.placeholder.forgetPassword") : ""}
                      value={passwordValue ?? ""}
                      onChange={onPasswordChange}
                      onBlur={onPasswordBlur}
                      inputRef={passwordRef}
                      inputProps={{
                        autoComplete: "password",
                      }}
                      InputProps={{
                        endAdornment: (
                          <IconButton
                            data-testid="button-toggle-show-password"
                            aria-label="Toggle show password"
                            onClick={handleToggleShowPassword}
                          >
                            {showPassword ? (
                              <VisibilityIcon aria-label="Showing password" />
                            ) : (
                              <VisibilityOffIcon aria-label="Hiding password" />
                            )}
                          </IconButton>
                        ),
                      }}
                      InputLabelProps={{
                        shrink: true,
                      }}
                    />
                    <Typography
                      data-testid="password-error"
                      variant="labelExtraSmallMedium"
                      textAlign="left"
                      color="error"
                    >
                      {error?.message || ""}
                    </Typography>
                  </Stack>
                )}
              />
              <Typography variant="labelExtraSmallMedium" textAlign="left" color="error">
                {errorMessage || ""}
              </Typography>

              <SubmitButtonStyle type="submit" data-testid="button-login" disabled={isDisableSubmit}>
                {translation("login.button.login")}
              </SubmitButtonStyle>
            </form>
            <ForgetPassword
              variant="bodySmallMedium"
              textTransform={"none"}
              color="text.primary"
              onClick={() => router.push(NAVIGATION.FORGET_PASSWORD)}
            >
              {translation("login.button.forgetPassword")}
            </ForgetPassword>
          </Container>
        </Stack>
        <ConsentFooter direction="row" spacing="8px">
          <Link href={NAVIGATION.PUBLIC_TERMS_AND_CONDITIONS} target="_blank">
            <Typography variant="labelExtraSmallMedium" data-testid="button-term-and-conditions">
              {translation("login.button.termsAndConditions")}
            </Typography>
          </Link>
          <Link href={NAVIGATION.PUBLIC_PRIVACY_POLICY} target="_blank">
            <Typography variant="labelExtraSmallMedium" data-testid="button-privacy-policy">
              {translation("login.button.privacyPolicy")}
            </Typography>
          </Link>
        </ConsentFooter>
      </Stack>
    </>
  );
};

export default LoginModule;
