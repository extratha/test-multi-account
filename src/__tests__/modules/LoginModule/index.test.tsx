import React from 'react';
import { act, waitFor, render } from '../../testUtils';
import LoginModule from '@/modules/LoginModule';
import { webPaths } from '@/constant/webPaths';
import * as cookiesNext from 'cookies-next';
// Import the module to mock
import * as nextRouter from 'next/router';

// Mock the module
jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: jest.fn(),
}));

describe('LoginModule', () => {
  beforeEach(() => {
    // Mock getCookie function to return a token
    jest.spyOn(cookiesNext, 'getCookie').mockReturnValue('accessToken');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('redirects to home page if accessToken exists', () => {
    // Mock router.replace function
    const replaceMock = jest.fn();
    (nextRouter.useRouter as any).mockReturnValue({ replace: replaceMock });
    jest.spyOn(cookiesNext, 'getCookie').mockReturnValue('TOKENTOKENTOKEN' as any);

    render(<LoginModule />);

  });

  it('renders login form if accessToken does not exist', () => {
    // Mock router.replace function
    const replaceMock = jest.fn();
    (nextRouter.useRouter as any).mockReturnValue({ replace: replaceMock });

    // Mock getCookie function to return null
    jest.spyOn(cookiesNext, 'getCookie').mockReturnValue(null as any);

    render(<LoginModule />);

  });
  it('to match snap', async () => {
    const { asFragment } = await act(async () =>
      render(
        <LoginModule />,
      ),
    );
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot()
    });
  })
});