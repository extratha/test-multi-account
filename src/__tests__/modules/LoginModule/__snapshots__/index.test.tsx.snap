// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModule to match snap 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-19hqra5-MuiStack-root"
  >
    <div
      class="MuiStack-root css-tvwe8b-MuiStack-root"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; position: absolute; z-index: -1; width: 100%; height: 100%; object-fit: cover;"
        width="40"
      />
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; margin: 40vh auto 0px;"
        width="40"
      />
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; margin: auto auto 3rem;"
        width="40"
      />
    </div>
    <div
      class="MuiStack-root css-1coyx4v-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1bi8i7x-MuiStack-root"
      >
        <div
          class="MuiStack-root css-nen11g-MuiStack-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-headlineSmallSemiBold css-eyjvzl-MuiTypography-root"
          >
            เข้าสู่ระบบ
          </h2>
          <h2
            class="MuiTypography-root MuiTypography-headlineLargeSemiBold css-10hprht-MuiTypography-root"
          >
            Cariva Playground
          </h2>
          <form>
            <div
              class="MuiStack-root css-x6fbp7-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-8qddx0-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-249s3f-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="email"
                    name="email"
                    placeholder="อีเมล"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <span
                class="MuiTypography-root MuiTypography-bodyLarge css-1ape98t-MuiTypography-root"
                data-testid="error"
              />
            </div>
            <div
              class="MuiStack-root css-x6fbp7-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-nz95qv-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1qnkz60-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="password"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="password"
                    name="password"
                    placeholder="รหัสผ่าน"
                    type="password"
                    value=""
                  />
                  <a>
                    <button
                      aria-label="Toggle show password"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="button-toggle-show-password"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="Hiding password"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="VisibilityOffIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7M2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2m4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <span
                class="MuiTypography-root MuiTypography-bodyLarge css-1ape98t-MuiTypography-root"
                data-testid="error"
              />
            </div>
            <span
              class="MuiTypography-root MuiTypography-bodyLarge css-1ape98t-MuiTypography-root"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-hgkxr1-MuiButtonBase-root-MuiButton-root"
              data-testid="button-login"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiTypography-root MuiTypography-labelExtraLargeSemiBold css-2pslf3-MuiTypography-root"
              >
                เข้าสู่ระบบ
              </span>
            </button>
          </form>
          <h3
            class="MuiTypography-root MuiTypography-titleSmallSemiBold css-1mh7n3j-MuiTypography-root"
          >
            ลืมรหัสผ่าน
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
